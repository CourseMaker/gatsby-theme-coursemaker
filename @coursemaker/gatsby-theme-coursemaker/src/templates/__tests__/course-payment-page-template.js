import React from 'react';
import renderer from 'react-test-renderer';
import CoursePaymentPage from '../course-payment-page-template';
import pageContextDouble from '../__mocks__/course-payment-page-double';

describe('CoursePaymentPage', () => {
    it('renders one-time purchase text correctly', () => {
        // Created using the query from Header.js
        const pageContextTestSample = pageContextDouble;
        const tree = renderer.create(<CoursePaymentPage pageContext={pageContextTestSample} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('CoursePaymentPage', () => {
    it('displays membership purchase text', () => {
        // Created using the query from Header.js
        const pageContextTestSample = pageContextDouble;
        pageContextTestSample.course.price_info = {
            active: true,
            currency: 'USD',
            name: 'Testing and Monitoring Price',
            product_type: 'school_membership',
            product_type_readable: 'Testing and Monitoring ML Models',
            recurring_interval: null,
            recurring_interval_count: null,
            unit_amount: 1995,
            unit_amount_readable: '$19.95',
            courses: [
                {
                    id: '102',
                },
            ],
        };
        const tree = renderer.create(<CoursePaymentPage pageContext={pageContextTestSample} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('CoursePaymentPage', () => {
    it('handles inactive price', () => {
        // Created using the query from Header.js
        const pageContextTestSample = pageContextDouble;
        pageContextTestSample.course.price_info = {
            active: false,
            currency: 'USD',
            name: 'Testing and Monitoring Price',
            product_type: 'school_membership',
            product_type_readable: 'Testing and Monitoring ML Models',
            recurring_interval: null,
            recurring_interval_count: null,
            unit_amount: 1995,
            unit_amount_readable: '$19.95',
            courses: [
                {
                    id: '102',
                },
            ],
        };
        const tree = renderer.create(<CoursePaymentPage pageContext={pageContextTestSample} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
