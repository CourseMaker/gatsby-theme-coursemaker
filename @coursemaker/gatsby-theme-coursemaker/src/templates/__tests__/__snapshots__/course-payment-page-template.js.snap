// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursePaymentPage displays membership purchase text 1`] = `
Array [
  <header
    className="sticky top-0 z-20 shadow-md text-white bg-blue-500"
  >
    <div
      className="container mx-auto"
    >
      <div
        className="flex flex-wrap items-center"
      >
        <a
          aria-current="page"
          className="inline-flex items-center h-20 text-lg md:h-24 logo"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          <span
            className="font-semibold"
          >
            dsacademy2
          </span>
        </a>
        <nav
          className="items-center order-3 w-full ml-auto list-none lg:flex lg:space-y-0 space-y-3 
						lg:space-x-6 lg:order-1 lg:w-auto py-8 lg:py-0 border-t lg:border-t-0 border-white border-opacity-25 
						hidden"
        >
          <li>
            <a
              aria-current="page"
              className=""
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li>
            <a
              className=""
              href="/courses"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              My Courses
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              className=""
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
            >
              Logout
            </a>
          </li>
        </nav>
        <button
          className="block w-8 h-6 ml-auto cursor-pointer lg:hidden burger-menu 
						"
          onClick={[Function]}
          tabIndex={0}
        >
          <div
            className="bar"
          />
          <div
            className="bar"
          />
          <div
            className="bar"
          />
        </button>
      </div>
    </div>
  </header>,
  <main>
    <section
      className="py-16 text-center md:py-24"
    >
      <div
        className="container"
      >
        <div
          className="mx-auto lg:w-10/12"
        >
          <h1
            className="mb-4"
          >
            Testing and Monitoring Machine Learning Models
          </h1>
          <h3
            className="mb-4"
          >
            
             
            
          </h3>
          <div>
            <button
              className="btn text-white bg-blue-500 text-lg btn-lg"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
            >
              Purchase Course
            </button>
          </div>
        </div>
      </div>
    </section>
  </main>,
  <div
    className="CookieConsent"
    style={
      Object {
        "alignItems": "baseline",
        "background": "#353535",
        "bottom": "0",
        "color": "white",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "left": "0",
        "position": "fixed",
        "width": "100%",
        "zIndex": "999",
      }
    }
  >
    <div
      className=""
      style={
        Object {
          "flex": "1 0 300px",
          "margin": "15px",
        }
      }
    >
      This site uses cookies ...
    </div>
    <div
      className=""
    >
      <button
        aria-label="Accept cookies"
        className=""
        id="rcc-confirm-button"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffd42d",
            "border": "0",
            "borderRadius": "0px",
            "boxShadow": "none",
            "color": "black",
            "cursor": "pointer",
            "flex": "0 0 auto",
            "margin": "15px",
            "padding": "5px 10px",
          }
        }
      >
        Accept
      </button>
    </div>
  </div>,
  <div
    className="Toastify"
  />,
  <footer
    className="py-8 text-sm"
  >
    <div
      className="container"
    >
      <div
        className="flex justify-between"
      >
        <p
          className="text-center text-gray-500"
        >
          Powered by
          <a
            className="font-bold text-blue-500"
            href="https://coursemaker.org"
          >
             
            CourseMaker
          </a>
        </p>
        <div
          className="flex ml-auto space-x-5 text-blue-500"
        >
          <a
            href="/terms"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Terms
          </a>
          <a
            href="/privacy"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Privacy
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`CoursePaymentPage handles inactive price 1`] = `
Array [
  <header
    className="sticky top-0 z-20 shadow-md text-white bg-blue-500"
  >
    <div
      className="container mx-auto"
    >
      <div
        className="flex flex-wrap items-center"
      >
        <a
          aria-current="page"
          className="inline-flex items-center h-20 text-lg md:h-24 logo"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          <span
            className="font-semibold"
          >
            dsacademy2
          </span>
        </a>
        <nav
          className="items-center order-3 w-full ml-auto list-none lg:flex lg:space-y-0 space-y-3 
						lg:space-x-6 lg:order-1 lg:w-auto py-8 lg:py-0 border-t lg:border-t-0 border-white border-opacity-25 
						hidden"
        >
          <li>
            <a
              aria-current="page"
              className=""
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li>
            <a
              className=""
              href="/courses"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              My Courses
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              className=""
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
            >
              Logout
            </a>
          </li>
        </nav>
        <button
          className="block w-8 h-6 ml-auto cursor-pointer lg:hidden burger-menu 
						"
          onClick={[Function]}
          tabIndex={0}
        >
          <div
            className="bar"
          />
          <div
            className="bar"
          />
          <div
            className="bar"
          />
        </button>
      </div>
    </div>
  </header>,
  <main>
    <section
      className="py-16 text-center md:py-24"
    >
      <div
        className="container"
      >
        <div
          className="mx-auto lg:w-10/12"
        >
          <h1
            className="mb-4"
          >
            Testing and Monitoring Machine Learning Models
          </h1>
          <h3
            className="mb-4"
          >
            
             
            
          </h3>
          <div>
            <button
              className="btn text-white bg-blue-500 text-lg btn-lg"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
            >
              Purchase Course
            </button>
          </div>
        </div>
      </div>
    </section>
  </main>,
  <div
    className="CookieConsent"
    style={
      Object {
        "alignItems": "baseline",
        "background": "#353535",
        "bottom": "0",
        "color": "white",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "left": "0",
        "position": "fixed",
        "width": "100%",
        "zIndex": "999",
      }
    }
  >
    <div
      className=""
      style={
        Object {
          "flex": "1 0 300px",
          "margin": "15px",
        }
      }
    >
      This site uses cookies ...
    </div>
    <div
      className=""
    >
      <button
        aria-label="Accept cookies"
        className=""
        id="rcc-confirm-button"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffd42d",
            "border": "0",
            "borderRadius": "0px",
            "boxShadow": "none",
            "color": "black",
            "cursor": "pointer",
            "flex": "0 0 auto",
            "margin": "15px",
            "padding": "5px 10px",
          }
        }
      >
        Accept
      </button>
    </div>
  </div>,
  <div
    className="Toastify"
  />,
  <footer
    className="py-8 text-sm"
  >
    <div
      className="container"
    >
      <div
        className="flex justify-between"
      >
        <p
          className="text-center text-gray-500"
        >
          Powered by
          <a
            className="font-bold text-blue-500"
            href="https://coursemaker.org"
          >
             
            CourseMaker
          </a>
        </p>
        <div
          className="flex ml-auto space-x-5 text-blue-500"
        >
          <a
            href="/terms"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Terms
          </a>
          <a
            href="/privacy"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Privacy
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`CoursePaymentPage renders one-time purchase text correctly 1`] = `
Array [
  <header
    className="sticky top-0 z-20 shadow-md text-white bg-blue-500"
  >
    <div
      className="container mx-auto"
    >
      <div
        className="flex flex-wrap items-center"
      >
        <a
          aria-current="page"
          className="inline-flex items-center h-20 text-lg md:h-24 logo"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          <span
            className="font-semibold"
          >
            dsacademy2
          </span>
        </a>
        <nav
          className="items-center order-3 w-full ml-auto list-none lg:flex lg:space-y-0 space-y-3 
						lg:space-x-6 lg:order-1 lg:w-auto py-8 lg:py-0 border-t lg:border-t-0 border-white border-opacity-25 
						hidden"
        >
          <li>
            <a
              aria-current="page"
              className=""
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li>
            <a
              className=""
              href="/courses"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              My Courses
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              className=""
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={Object {}}
            >
              Logout
            </a>
          </li>
        </nav>
        <button
          className="block w-8 h-6 ml-auto cursor-pointer lg:hidden burger-menu 
						"
          onClick={[Function]}
          tabIndex={0}
        >
          <div
            className="bar"
          />
          <div
            className="bar"
          />
          <div
            className="bar"
          />
        </button>
      </div>
    </div>
  </header>,
  <main>
    <section
      className="py-16 text-center md:py-24"
    >
      <div
        className="container"
      >
        <div
          className="mx-auto lg:w-10/12"
        >
          <h1
            className="mb-4"
          >
            Testing and Monitoring Machine Learning Models
          </h1>
          <h3
            className="mb-4"
          >
            One-time purchase of: 
             
            $19.95
          </h3>
          <div>
            <button
              className="btn text-white bg-blue-500 text-lg btn-lg"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
            >
              Purchase Course
            </button>
          </div>
        </div>
      </div>
    </section>
  </main>,
  <div
    className="CookieConsent"
    style={
      Object {
        "alignItems": "baseline",
        "background": "#353535",
        "bottom": "0",
        "color": "white",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "left": "0",
        "position": "fixed",
        "width": "100%",
        "zIndex": "999",
      }
    }
  >
    <div
      className=""
      style={
        Object {
          "flex": "1 0 300px",
          "margin": "15px",
        }
      }
    >
      This site uses cookies ...
    </div>
    <div
      className=""
    >
      <button
        aria-label="Accept cookies"
        className=""
        id="rcc-confirm-button"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffd42d",
            "border": "0",
            "borderRadius": "0px",
            "boxShadow": "none",
            "color": "black",
            "cursor": "pointer",
            "flex": "0 0 auto",
            "margin": "15px",
            "padding": "5px 10px",
          }
        }
      >
        Accept
      </button>
    </div>
  </div>,
  <div
    className="Toastify"
  />,
  <footer
    className="py-8 text-sm"
  >
    <div
      className="container"
    >
      <div
        className="flex justify-between"
      >
        <p
          className="text-center text-gray-500"
        >
          Powered by
          <a
            className="font-bold text-blue-500"
            href="https://coursemaker.org"
          >
             
            CourseMaker
          </a>
        </p>
        <div
          className="flex ml-auto space-x-5 text-blue-500"
        >
          <a
            href="/terms"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Terms
          </a>
          <a
            href="/privacy"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Privacy
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
